SU2_CFD/src/solvers/CIncNSSolver.cpp

```diff
@@ -97,6 +98,43 @@ void CIncNSSolver::Preprocessing(CGeometry *geometry, CSolver **solver_container
 
   ComputeVorticityAndStrainMag<1>(*config, iMesh);
 
+  /*--- Compute recovered pressure and temperature for streamwise periodic flow ---*/
+  if (config->GetKind_Streamwise_Periodic() != NONE) {
+
+    /*--- Define and initialize helping variables ---*/
+    su2double dot_product, Pressure_Recovered, Temperature_Recovered;
+
+    const su2double delta_p  = config->GetStreamwise_Periodic_PressureDrop() / config->GetPressure_Ref();
+
+    /*--- Reference node on inlet periodic marker to compute relative distance along periodic translation vector. ---*/
+    const su2double* ReferenceNode = geometry->GetStreamwise_Periodic_RefNode();
+
+    /*--- Compute square of the distance between the 2 periodic surfaces. ---*/
+    const su2double norm2_translation = GeometryToolbox::SquaredNorm(nDim, config->GetPeriodic_Translation(0));
+
+    /*--- Compute recoverd pressure and temperature for all points ---*/
+    for (auto iPoint = 0ul; iPoint < nPoint; iPoint++) {
+
+      /*--- First, compute helping terms based on relative distance (0,l) between periodic markers ---*/
+      dot_product = 0.0;
+      for (unsigned short iDim = 0; iDim < nDim; iDim++)
+        dot_product += fabs( (geometry->nodes->GetCoord(iPoint,iDim) - ReferenceNode[iDim]) * config->GetPeriodic_Translation(0)[iDim]);
+
+      /*--- Second, substract/add correction from reduced pressure/temperature to get recoverd pressure/temperature ---*/
+      Pressure_Recovered = nodes->GetSolution(iPoint, 0) - delta_p / norm2_translation * dot_product;
+      nodes->SetStreamwise_Periodic_RecoveredPressure(iPoint, Pressure_Recovered);
+
+      /*--- InnerIter > 0 as otherwise MassFlow in the denominator would be zero ---*/
+      if (energy && InnerIter > 0) {
+        Temperature_Recovered  = nodes->GetSolution(iPoint, nDim+1);
+        Temperature_Recovered += Streamwise_Periodic_IntegratedHeatFlow / (Streamwise_Periodic_MassFlow * nodes->GetSpecificHeatCp(iPoint) * norm2_translation) * dot_product;
+        nodes->SetStreamwise_Periodic_RecoveredTemperature(iPoint, Temperature_Recovered);
+      }
+    }
+
+    /*--- Compute the integrated Heatflux Q into the domain, and massflow over periodic markers ---*/
+    GetStreamwise_Periodic_Properties(geometry, config, iMesh);
```

Comment:
SU2_OMP_MASTER
...
SU2_OMP_BARRIER

For the easy way out

Reply:
Done. 
SU2_OMP_BARRIER = #pragma omp barrier i.e. a synchronization point
SU2_OMP_MASTER = #pragma omp master i.e.  the next structured block is only executed by the master thread
Otherwise all threads would call and execute that function, right?

Reply:
Yup, barrier makes shared variables consistent across threads, in GetStreamwise_Periodic_Properties you write to shared variables, after the barrier all threads "see" the new values.